genesyscloud/journey_views/resource_genesyscloud_journey_views_schema.go

```diff
@@ -0,0 +1,213 @@
+			"id": {
+				Description: "The identifier for the element based on its type.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"source": {
+				Description: "The source for the element (e.g. IVR, Voice, Chat). Used for informational purposes only.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+		},
+	}
+	filtersResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Description:  "Boolean operation to apply to the provided predicates and clauses. Valid values: And.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"And"}, false),
+			},
+			"predicates": {
+				Description: "A filter on an element within a journey view.",
+				Type:        schema.TypeList,
+				Elem:        predicatesResource,
+				Optional:    true,
+			},
+		},
+	}
+	elementsResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "The unique identifier of the element within the elements list.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"name": {
+				Description: "The unique name of the element within the view.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"attributes": {
+				Description: "Attributes on an element in a journey view.",
+				Type:        schema.TypeList,
+				Required:    true,
+				Elem:        attributesResource,
+				MaxItems:    1,
+			},
+			"filter": {
+				Description: "A set of filters on an element within a journey view.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        filtersResource,
+				MaxItems:    1,
+			},
+			"followed_by": {
+				Description: "A list of JourneyViewLink objects, listing the elements downstream of this element.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        followedByResource,
+			},
+		},
+	}
+)
+
+func SetRegistrar(regInstance registrar.Registrar) {
+	regInstance.RegisterResource(resourceName, ResourceGroup())
+	/***
+	TODO: Add DataSource and Exporter once we are done with https://inindca.atlassian.net/browse/JM-109
+	regInstance.RegisterDataSource(resourceName, DataSourceGroup())
+	regInstance.RegisterExporter(resourceName, JourneyViewExporter())
+	***/
+}
+
+func ResourceGroup() *schema.Resource {
+	return &schema.Resource{
+		Description: "Genesys Cloud Directory JourneyView",
+
+		CreateContext: provider.CreateWithPooledClient(createJourneyView),
+		ReadContext:   provider.ReadWithPooledClient(readJourneyView),
+		UpdateContext: provider.UpdateWithPooledClient(updateJourneyView),
+		DeleteContext: provider.DeleteWithPooledClient(deleteJourneyView),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "JourneyView name.",
+				Type:        schema.TypeString,
+				Optional:    true,
```

Comment:
Are you sure name should be optional? This api might be different but most api's require name

Reply:
name is optional as of now. But yes I guess it's better to make it required, let me talk to my team so that we make the changes in the api